packages/salesforcedx-vscode-core/src/commands/util/useBetaDeployRetrieve.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  RegistryAccess,
+  registryData
+} from '@salesforce/source-deploy-retrieve';
+import * as vscode from 'vscode';
+
+import { sfdxCoreSettings } from '../../settings';
+
+export function useBetaDeployRetrieve(explorerPath: vscode.Uri[]): boolean {
```

Comment:
I think I once broke this piece of code by trying to retrieve a file that is not present in the Org. That library was returning an empty metadata component so all the object navigation we do here (e.g. `component.type.name`) was throwing errors.

Can you test if this is still valid and if so we need to handle that scenario.